### ISSUE TITLE
v2.2.0 Roadmap

### ISSUE THREAD

**Issue by Chalarangelo** [Chalarangelo/mini.css#74]
## Overview

The release of **v2.2.0** will change a lot of things around and it will also bring about some [breaking changes](#compatibility). A list of changes, along with their progress is shown below.

## Upcoming changes

- [x] Introduction of a `.drawer` (name is not finalized) component in `navigation`, allowing for slide-in menus and responsive navigation.
- [x] Introduction of fluid typography, effectively scaling the root element of the document, based on the screen size (uses `calc()`). This feature will most likely be opt-in, meaning that old flavors will not be affected, as the feature will be turned off by default.
- [x] The ordering of modules will be changed (`input_control` should come before `navigation` at least) to improve style cascading and gzip compression.
- [x] Re-evaluation of all `padding` and `margin` values, making sure that no oddities and artifacts exist and that everything works quite well with fluid typography.
- [x] The responsive margin and padding classes of the `utility` module will be revisited to make sure their `padding` and `margin` values are useful and `!important` declarations will be added to make sure they are applied (found a bug where they do not apply properly).
- [x] The `.alert` classes of the `contextual` module will be retired (I will either move them to their own component module as an extra or flag-disable them). A `.toast` class will take their place, making it easier to show toast messages on mobile devices.
- [x] Classes similar to the `.urgent` and `.critical` alerts will be added as `.card` classes.
- [x] The `.inverse` `.card` class will be removed, as it serves no practical purpose except to showcase the specific card mixin.
- [x] The `.nano` class will be removed from the `progress` module, as it serves no real purpose right now.
- [x] The `.close` icon class will be revisited and redesigned to utilize a textual `::before` pseudo-element with simpler rules, making it more lightweight. Also, extra rules might be added to make it compatible with `<label>` elements and the new `.drawer` component.
- [x] Generic shadow classes will be revisited, most likely removing most of them and adding a single generic shadow class for elements (the `.shadow-none` class will be probably gone as well).

## Compatibility

**v2.2.0** will introduce a few new components and remove some older ones, in order to improve functionality and de-bloat the framework somewhat. Refer to the above list of changes for more details.

## Release date

Worst case scenario is that **v2.2.0** will take until mid-June, but that is highly unlikely. The most likely scenario is that it will be available sometime towards the end of May. Stay tuned for more updates!

**Comment by Chalarangelo** [comment]
## Update

Due to a super secret project I want to work on, I am sort of rushing the update so that it can be available as soon as possible. So far, it's coming along nicely and it's about 60% done already. Most of the new features had been tested before starting work on the update itself, so many of the quirks have already been ironed out. In order to actually get the update out as soon as possible, some compromises might be made, mainly in terms of changing the page layout and documentation (some navigational updates might be postponed until next week), but the quality of the framework is the same as usual. Apart from that, the two *custom* flavors, `Nord` and `Sucroa` are not fully updated (due to the changes in certain components such as `.alert`s) and they will either be updated by their authors/maintainers in a later patch or I will get around to fixing them sometime in June. They are still perfectly fine and work great, it's just that some of the new features and changes will not hit them until the end of the month or so. Sorry about that.

## Estimated release date

Originally, I was going for a release around the end of the month, but right now I am aiming to release the update either tomorrow or the day after. Don't take my word for it, though, just yet!

**Comment by mistakenelf** [comment]
Just wanna say. Absolutely loving this project. I'm curious what this super secret project is is now.

**Comment by Chalarangelo** [comment]
## Update

Everything that was promised is now (mostly) complete. Due to the fact that I have some extra time left, I will try to update @tphecca's **Nord** flavor to use the latest components (the only component where I will arbitrarily pick values from the theme is the `.toast`), as well as look into @AngieDaskalakis's **Sucroa** flavor file to see if I can update certain things to the latest version. If any time is left over, I will try to bump at least **Nord** to the `rem` sizing (as the values of all the `margin`s and `padding`s are the same as the **Default** flavor), so that it is more up-to-date. Besides these changes, I will update the doc pages to utilize the `.drawer` to make the website more mobile-friendly. A new template will be coming to showcase the new `.drawer` component, as well as updates to existing ones. 

As for the super secret project, @knipferrc, I have to say that it has to do with a more extensive version of templates and *might* ~definitely~ have something to do with Codepen. You might see something on the website in the next 2-3 days following the update. Also, there is another super secret project coming sometime later down the line, involving React (but I'm not sure I will actually go through with it to be honest, so don't get your hopes too high just yet).

## Release date

Hopefully in the next 12 hours!

## Notes

**v2.2.0** will be the foundation for the first **LTS** release (which will be something like **v2.2.4**, as soon as any bugs are squashed and I'm sure of its stability), so expect the featureset to stay pretty much the same for the next 6 or so months. A couple of things might be coming, such as Android-style toggles and a very basic modal, but don't take my word for it.

**Comment by Chalarangelo** [comment]
As far as the **Nord** flavor is concerned, refer to 38bfc6e for a list of changes to the flavor. I took the liberty of updating to **v2.2.0** as it is quite a big update, so at least it will work similarly to the **Default** flavor.

**Comment by Chalarangelo** [comment]
As far as the **Sucroa** flavor is concerned, refer to 076fd2d for a list of changes to the flavor. I updated to **v2.2.0**, hopefully keeping in line with its original style.

**Comment by Chalarangelo** [comment]
As far as the **Sucroa** flavor is concerned, refer to 076fd2d for a list of changes to the flavor. I updated to **v2.2.0**, hopefully keeping in line with its original style.

**Comment by Chalarangelo** [comment]
@knipferrc I just wanted to say that the super secret project is now part of the [Templates page](http://minicss.org/templates.html) in case you missed it. Hope you like it! :wink:
